File: apps/agent/app/llm/factory.py
"""LLM provider factory — resolves provider based on configuration."""

from app.config import Settings
from app.llm.base import LLMProvider
from app.llm.mock import MockLLMProvider
from app.llm.claude import ClaudeLLMProvider
from app.llm.gemini import GeminiLLMProvider


def get_llm_provider(settings: Settings) -> LLMProvider:
    """Return configured LLM provider instance."""
    provider = settings.llm_provider.lower().strip()

    if provider == "mock":
        return MockLLMProvider()
    elif provider == "claude":
        return ClaudeLLMProvider(api_key=settings.anthropic_api_key)
    elif provider == "gemini":
        return GeminiLLMProvider(api_key=settings.gemini_api_key)
    else:
        raise ValueError(
            f"Unknown LLM_PROVIDER '{provider}'. Supported providers: 'mock', 'claude', 'gemini'."
        )

File: apps/agent/src/llm/factory.ts
import { config } from '../config';
import type { LLMProvider } from './base';
import { MockLLMProvider } from './mock';
import { ClaudeLLMProvider } from './claude';
import { GeminiLLMProvider } from './gemini';

export function getLLMProvider(provider?: string): LLMProvider {
  const selected = (provider || config.LLM_PROVIDER).toLowerCase();

  switch (selected) {
    case 'mock':
      return new MockLLMProvider();
    case 'claude':
      return new ClaudeLLMProvider();
    case 'gemini':
      return new GeminiLLMProvider();
    default:
      throw new Error(`Unknown LLM provider: '${selected}'. Available: mock, claude, gemini`);
  }
}
